tests/Auth/AuthMiddlewareTest.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Yiisoft\Yii\Web\Tests\Auth;
+
+use Nyholm\Psr7\Factory\Psr17Factory;
+use Nyholm\Psr7\ServerRequest;
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Yiisoft\Yii\Web\Auth\AuthInterface;
+use Yiisoft\Yii\Web\Auth\AuthMiddleware;
+use PHPUnit\Framework\TestCase;
+use Yiisoft\Yii\Web\User\IdentityInterface;
+
+class AuthMiddlewareTest extends TestCase
+{
+    /** @var ResponseFactoryInterface */
+    private $responseFactory;
+
+    /** @var AuthInterface */
+    private $authenticator;
+
+    protected function setUp()
+    {
+        $this->responseFactory = new Psr17Factory();
+        $this->authenticator = $this->createMock(AuthInterface::class);
+    }
+
+    /**
+     * @test
+     */
+    public function shouldAuthenticateAndSetAttribute()
+    {
+        $request = new ServerRequest('GET', '/');
```

Comment:
Don't know, but all other test have been written with mocks here :/ @samdark ?

Reply:
Not all :) https://github.com/yiisoft/yii-web/pull/137#discussion_r341182954

Reply:
It's fine.